tests/phpunit/StorageExtractorTest.php

```diff
@@ -0,0 +1,326 @@
+        $app->run();
+    }
+
+    public function testTableFilter(): void
+    {
+        $temp = new Temp('ex-storage');
+        $temp->initRunFolder();
+        $fs = new Filesystem();
+        $fs->dumpFile($temp->getTmpFolder() . '/tmp.csv', "\"id\",\"foo\"\n\"1\",\"aa\"\n\"2\",\"bb\"\n\"3\",\"cc\"\n");
+        $csv = new CsvFile($temp->getTmpFolder() . '/tmp.csv');
+        $this->client->createTable(getenv('KBC_TEST_BUCKET'), 'some-table-2', $csv, ['primaryKey' => 'id']);
+        $fs->dumpFile($temp->getTmpFolder() . '/tmp.csv', "\"id\",\"bar\"\n\"1\",\"x\"\n\"2\",\"y\"\n\"3\",\"z\"\n");
+        $csv = new CsvFile($temp->getTmpFolder() . '/tmp.csv');
+        $this->client->createTable(getenv('KBC_TEST_BUCKET'), 'some-table-3', $csv, ['primaryKey' => 'id']);
+
+        $configFile = [
+            'action' => 'run',
+            'parameters' => [
+                '#token' => getenv('KBC_TEST_TOKEN'),
+                'url' => getenv('KBC_TEST_URL'),
+                'tables' => ['some-table-3'],
+            ],
+        ];
+        $baseDir = $temp->getTmpFolder();
+        $fs->dumpFile($baseDir . '/config.json', \GuzzleHttp\json_encode($configFile));
+        putenv('KBC_DATADIR=' . $baseDir);
+        $app = new Component(new NullLogger());
+        $app->run();
+        self::assertFileExists($baseDir . '/out/tables/some-table-3.csv');
+        self::assertFileNotExists($baseDir . '/out/tables/some-table-2.csv');
+        $csv = new CsvFile($baseDir . '/out/tables/some-table-3.csv');
+        $rows = iterator_to_array($csv);
+        sort($rows);
+        self::assertEquals(
+            [
+                ['1', 'x'],
+                ['2', 'y'],
+                ['3', 'z'],
+                ['id', 'bar'],
+            ],
+            $rows
+        );
+        self::assertFileNotExists($baseDir . '/out/tables/some-table-2.csv.manifest');
+        self::assertFileExists($baseDir . '/out/tables/some-table-3.csv.manifest');
+        $data = json_decode(file_get_contents($baseDir . '/out/tables/some-table-3.csv.manifest'), true);
+        self::assertEquals([], $data);
+    }
+
+    public function testTableInvalidFilter(): void
+    {
+        $temp = new Temp('ex-storage');
+        $temp->initRunFolder();
+        $fs = new Filesystem();
+        $configFile = [
+            'action' => 'run',
+            'parameters' => [
+                '#token' => getenv('KBC_TEST_TOKEN'),
+                'url' => getenv('KBC_TEST_URL'),
+                'tables' => ['non-existent-table'],
+            ],
+        ];
+        $baseDir = $temp->getTmpFolder();
+        $fs->dumpFile($baseDir . '/config.json', \GuzzleHttp\json_encode($configFile));
+        putenv('KBC_DATADIR=' . $baseDir);
+        $app = new Component(new NullLogger());
+        self::expectExceptionMessage('The table non-existent-table not found in the bucket in.c-ex-storage');
```

Comment:
Tady by byly taky fajn uvozovky v error message. Třeba když bude název tabulky "is" nebo "was" :)

Reply:
no tak to si fixnes sam tady https://github.com/keboola/connection/issues/1354 protoze to je sproste preposlanej error :)